alerting/src/main/kotlin/org/opensearch/alerting/BucketLevelMonitor.kt

```diff
@@ -0,0 +1,350 @@
+package org.opensearch.alerting
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import org.apache.logging.log4j.LogManager
+import org.opensearch.alerting.elasticapi.InjectorContextElement
+import org.opensearch.alerting.model.InputRunResults
+import org.opensearch.alerting.model.BucketLevelTrigger
+import org.opensearch.alerting.model.Alert
+import org.opensearch.alerting.model.AlertingConfigAccessor
+import org.opensearch.alerting.model.ActionRunResult
+import org.opensearch.alerting.model.BucketLevelTriggerRunResult
+import org.opensearch.alerting.model.Monitor
+import org.opensearch.alerting.model.MonitorRunResult
+import org.opensearch.alerting.model.action.ActionExecutionScope
+import org.opensearch.alerting.model.action.Action
+import org.opensearch.alerting.model.action.AlertCategory
+import org.opensearch.alerting.model.action.PerAlertActionScope
+import org.opensearch.alerting.model.action.PerExecutionActionScope
+import org.opensearch.alerting.script.BucketLevelTriggerExecutionContext
+import org.opensearch.alerting.script.TriggerExecutionContext
+import org.opensearch.alerting.util.getActionExecutionPolicy
+import org.opensearch.alerting.util.getBucketKeysHash
+import org.opensearch.alerting.util.getCombinedTriggerRunResult
+import org.opensearch.alerting.util.isAllowed
+import org.opensearch.common.Strings
+import java.time.Instant
+
+class BucketLevelMonitor: MonitorRunner {
```

Comment:
NP: Can we call these classes something like BucketLevelMonitor**Runner** instead to disambiguate from the data model objects?

Also you can make all of these runner types `objects` so they're singletons if you don't expect there to be more than one instance of it.